src/os/os.c

```diff
@@ -535,6 +548,28 @@ int os_thread_create(os_thread_t *thread, const char *name, int priority,
     add_thread_to_task_list(task_new);
 
     return 0;
+#elif defined(ESP32)
+    if (priority == 0)
+    {
+        priority = configMAX_PRIORITIES / 2;
+    }
+    else if (priority >= configMAX_PRIORITIES)
+    {
+        priority = configMAX_PRIORITIES - 1;
+    }
+    
+    if (stack_size == 0)
+    {
+        stack_size = 2048;
+    }
+    ThreadPriv *priv = malloc(sizeof(ThreadPriv));
+    priv->entry = start_routine;
+    priv->arg = arg;
+    LOG(VERBOSE, "Creating task %s (priority: %d, stack: %d)",
+        name, priority, stack_size);
+    xTaskCreatePinnedToCore(os_thread_start, name, stack_size, priv, priority,
```

Comment:
esp32 docs say to avoid the Generic API and instead use xTaskCreate. It should work the same though since it is created as a floating task. One problem with the current FreeRTOS masked code is that it stops the task scheduler which could lead to other complications on the platform. I'll give it a try and see if it works though and mask off the pieces of the generic FreeRTOS code used here if it causes any issues (mainly the scheduler interactions, the IDLE thread on the ESP32 handles the task cleanup automatically on vTaskDelete)